### PR TITLE
.github: correct the regex in milestone workflow

### PR DESCRIPTION
**Pull request by tchaikov**
also use pull_request_target event so the action is run in the
context of the base of the pull request. this helps us to overcome
the "Resource not accessible by integration" issue where the action
is run in the context of the pull request.

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
tested at https://github.com/tchaikov/ceph/pull/8

File: .github/milestone.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-  - "luminous"
-  - "nautilus"
-  - "octopus"
-  - "pacific"
-  - "quincy"
+  - "(luminous)"
+  - "(nautilus)"
+  - "(octopus)"
+  - "(pacific)"
+  - "(quincy)"
```
Comment (epuertat) [inline .github/milestone.yml:6]: Not sure how this regex works, but could we make (if it makes sense) `master` match 'quincy' (whenever I see a `master` PR merged I can never directly guess which release it was eventually merged into).

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -9,3 +9,10 @@ jobs:
         with:
           sync-labels: ''
           repo-token: "${{ secrets.GITHUB_TOKEN }}"
+  milestone:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: iyu/actions-milestone@v1
+        with:
+          configuration-path: .github/milestone.yml
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (epuertat) [inline .github/workflows/labeler.yml:18]: Great idea! Feel free to change the `labeler.yml` filename and workflow name ('Pull Request Labeler') to something more comprehensive (`pull_request_setup.yml`?).

On adding this as separate jobs (as here) vs. steps, as soon as we don't hit the [GH free plan limit (2,000 minute/month)](https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions), we can still go with jobs (they'll be faster as they'll run in parallel).

File: .github/workflows/labeler.yml [hunk 3]
```diff
@@ -9,3 +9,10 @@ jobs:
         with:
           sync-labels: ''
           repo-token: "${{ secrets.GITHUB_TOKEN }}"
+  milestone:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: iyu/actions-milestone@v1
```
Comment (epuertat) [inline .github/workflows/labeler.yml:15]: I forgot to change it in the `users: actions/labeler@main` line too (already [did it for the needs-rebase workflow](https://github.com/ceph/ceph/blob/master/.github/workflows/needs-rebase.yml#L14)), but due to [security concerns](https://julienrenaux.fr/2019/12/20/github-actions-security-risk/) (as the actions have access to our GH Token), it'be safer if we pin the actions to specific commits (and perhaps we should perform some basic checking in the action source-code to ensure it does not log/pushes the token anywhere).

**Review by epuertat — CHANGES_REQUESTED** [review]
Nice improvement @tchaikov ! I have a couple of change requests over there.

File: .github/milestone.yml (hunk 1 above)
Comment (tchaikov) [reply .github/milestone.yml:6]: the action tries to match the head-branch with the regex, and if it matches, the matched group is used for the milestone. i cannot think of a regex which can make "master" match "quincy" at this moment.

File: .github/workflows/labeler.yml (hunk 2 above)
Comment (tchaikov) [reply .github/workflows/labeler.yml:18]: indeed. as it both adds labels *and* set milestone, i'd like to have a better and less misleading name than "Pull Request Labeler" without sacrificing  the information passed by the original name. also, to "label" something can be very general when one considers milestone yet another label. =)

but you know, to name something is difficult in computer science ...

File: .github/workflows/labeler.yml (hunk 3 above)
Comment (tchaikov) [reply .github/workflows/labeler.yml:15]: do you want me to add a commit to fix this?

File: .github/workflows/labeler.yml (hunk 2 above)
Comment (tchaikov) [reply .github/workflows/labeler.yml:18]: regarding to the quota, yes. that's the primary reason why i create a commit to merge these two actions. another reason is to have less "checks". IMHO, to complete the two of the checks is not my concern, we know, the bottle neck is always the "ceph API test" and the "make check" runs.

Comment (tchaikov) [reply .github/workflows/labeler.yml:18]: also, you feel strong on the naming, how about dropping the commit merging these two actions from this PR? so we can backport the milestone action to LTS branches first.

Comment (epuertat) [reply .github/workflows/labeler.yml:18]: > indeed. as it both adds labels _and_ set milestone, i'd like to have a better and less misleading name than "Pull Request Labeler" without sacrificing the information passed by the original name. also, to "label" something can be very general when one considers milestone yet another label. =)

So what about "Pull Request Configuration"? If I'm not wrong, each of the jobs will generate separate 'checks' as follows:
* Pull Request Configuration / labeler (pull_request_target)
* Pull Request Configuration / milestone (pull_request_target)

If you want to print a single check, you'll need to move both to a single job with 2 steps.

> but you know, to name something is difficult in computer science ...

And not only in computer science, also in science fic... I meant, astrophysics. I was recently listening a podcast about how [this new term (ichnoscale)](https://arxiv.org/pdf/2103.01536.pdf) was coined.

File: .github/workflows/labeler.yml (hunk 3 above)
Comment (tchaikov) [reply .github/workflows/labeler.yml]: the linked article is a very good reading, thank you! I also point the actions-milestone to a sha1 in the same commit.

File: .github/workflows/labeler.yml (hunk 2 above)
Comment (tchaikov) [reply .github/workflows/labeler.yml]: 🤣  updated and repushed

**Comment by tchaikov** [comment]
@epuertat could you take another look?

File: .github/workflows/labeler.yml (hunk 2 above)
Comment (tchaikov) [reply .github/workflows/labeler.yml]: @epuertat just came across https://docs.github.com/en/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow. i feels that "triage" is a better name than "configuration" or "setup".

Comment (epuertat) [reply .github/workflows/labeler.yml]: Makes a lot of sense. Like that! 👍🏼

**Review by epuertat — APPROVED** [review]
Ready to go! :D

**Pull request merged**